internal/controller/controllers/hypershiftagentserviceconfig_controller.go

```diff
@@ -463,12 +460,17 @@ func (hr *HypershiftAgentServiceConfigReconciler) cleanStaleSpokeAgentInstallCRD
 
 func (hr *HypershiftAgentServiceConfigReconciler) getAgentInstallCRDs(ctx context.Context, c client.Client) (*apiextensionsv1.CustomResourceDefinitionList, error) {
 	crds := &apiextensionsv1.CustomResourceDefinitionList{}
-	listOpts := []client.ListOption{
-		client.HasLabels{fmt.Sprintf("operators.coreos.com/assisted-service-operator.%s", hr.Namespace)},
-	}
-	if err := c.List(ctx, crds, listOpts...); err != nil {
+	if err := c.List(ctx, crds); err != nil {
 		return nil, pkgerror.Wrap(err, "Failed to list CRDs")
 	}
+
+	// Filter CRDs by 'agent-install.openshift.io' and 'extensions.hive.openshift.io' groups
+	// (to include our agentclusterinstalls hive extension)
+	agentInstallCRDs := funk.Filter(crds.Items, func(crd apiextensionsv1.CustomResourceDefinition) bool {
+		return crd.Spec.Group == "agent-install.openshift.io" || crd.Spec.Group == "extensions.hive.openshift.io"
```

Comment:
i think that we already have a const for that https://github.com/openshift/assisted-service/blob/3d6d97472d74c37a14fe0a9d0a4cc538dd28f004/api/v1beta1/groupversion_info.go#L28
and 
https://github.com/openshift/assisted-service/blob/3d6d97472d74c37a14fe0a9d0a4cc538dd28f004/api/hiveextension/v1beta1/groupversion_info.go#L28

Reply:
What guarantees that this logic will be always aligned with OLM's? 
I would keep the old code, and only if there are no labels present, take the other route.

In addition, take the implementation to a well documented function, so future maintainers will be able to quickly identify any issues of OLM-based installations start to behave differently than OCM's

Reply:
We are defining the group when creating the CRDs (e.g. [here](https://github.com/openshift/assisted-service/blob/3d6d97472d74c37a14fe0a9d0a4cc538dd28f004/config/crd/resources.yaml#L9)), so no need to fetch CRDs twice, it would work regardless of OLM.
I can indeed extract the filter logic to another func if it makes the code clearer.